modules/ui/org.eclipse.fx.ui.preferences/src/main/java/org/eclipse/fx/ui/preferences/page/FloatFieldEditor.java

```diff
@@ -25,35 +27,44 @@
  * A Field editor for float preferences.
  * </p>
  */
-public class FloatFieldEditor extends FieldEditor {
-	private final TextField textField;
-
+public class FloatFieldEditor extends FieldEditor<Float> {
 	private static final Logger LOGGER = LoggerCreator.createLogger(FloatFieldEditor.class);
+	
+	private final TextField textField;
+	private SimpleFloatProperty value = new SimpleFloatProperty(this, "value");
 
 	public FloatFieldEditor(String name, String label) {
 		super(name, label);
 		this.textField = new TextField();
 		getChildren().add(textField);
 
-		configureValidation(this.textField);
+		NodeDecorator.apply(this.textField, statusProperty());
+		
+		this.textField.textProperty().addListener((obs, old, newValue) -> {
+			try {
+				value.set(Float.parseFloat(newValue));
```

Comment:
Does JFace it also this way? parseFloat does not take care of locale informations (eg in german the floating-point separator is a `,`)

Reply:
No, JFace doesn't provide a Float editor at all :) (Only an Integer editor, which relies on Integer.parseInt(), via Integer.valueOf())